subgraph/src/fleek-nfa.ts

```diff
@@ -203,6 +206,41 @@ export function handleTokenRoleGranted(event: TokenRoleGrantedEvent): void {
   entity.transactionHash = event.transaction.hash;
 
   entity.save();
+
+  if (event.params.role === 1) {
+    // This is a new controller being added to a token.
+    // First we add the controller to the token's list of controllers.
+    // Then we create a new controller entity.
+
+    let token = Token.load(
+      Bytes.fromByteArray(Bytes.fromBigInt(event.params.tokenId))
+    );
+    let controller = Controller.load(event.params.toAddress);
+
+    if (!controller) {
+      // Create a new controller entity
+      log.debug('CONTROLLER IS GOING TO BE CREATED HERE.', []);
+      controller = new Controller(event.params.toAddress);
+    }
+
+    if (token !== null) {
+      let token_controllers = token.controllers;
+      if (!token_controllers) {
+        token_controllers = [];
+      }
+      token_controllers.push(event.params.toAddress);
+      token.controllers = token_controllers;
+    } else {
+      log.error(
+        'Controller access granted for tokenId {}. THE TOKEN DOES NOT EXIST. FAILED TO UPDATE THE TOKEN ENTITY.',
```

Comment:
Can we change the first sentence to `Handling controller access granted event for tokenId {}`? The way it is now makes it seem like it is this call that did the granting but it's really processing the event for it.

Reply:
Of course. Will push a commit for this.